Rollbar/Serialization/Json/JsonScrubber.cs

```diff
@@ -217,8 +217,39 @@ public static void ScrubJsonPath(JObject jsonData, string scrubPath, string scru
                 return;
             }
 
-            var jProperty = jsonData.SelectToken(scrubPath)?.Parent as JProperty;
-            jProperty?.Replace(new JProperty(jProperty.Name, scrubMask));
+            var jToken = jsonData.SelectToken(scrubPath);
+            if (jToken != null)
+            {
+                var jProperty = jToken.Parent as JProperty;
+                if (jProperty != null)
+                {
+                    jProperty.Replace(new JProperty(jProperty.Name, scrubMask));
+                    return;
+                }
+            }
+
+            //to deal with the possible dotted data element name we need to perform some acrobatics here:
+            int indxLimit = scrubPath.LastIndexOf('.');
+            int startingIndx = 0;
+            int dotIndx = scrubPath.IndexOf('.', startingIndx);
+            while(dotIndx > 0 && dotIndx < indxLimit)
```

Comment:
Nit: I believe there should be a whitespace inserted: `while (`. On a side note, is there a linter we can deploy to ensure style conformance?

Reply:
Fixed. Yes, we use SonurCloud with this SDK.

Reply:
The relevant StyleCop rule is `KeywordsMustBeSpacedCorrectly`, but either it isn't enabled or StyleCop isn't running at all in Sonar.